src/libs/sys-util-rs/src/numa.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::collections::HashMap;
+use std::fs::DirEntry;
+use std::io::Read;
+use std::path::Path;
+
+use lazy_static::lazy_static;
+
+use crate::cpu::split_cpus;
+use crate::sl;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Invalid CPU number {0}")]
+    InvalidCpu(u32),
+    #[error("Invalid node file name {0}")]
+    InvalidNodeFileName(String),
+    #[error("Can not read from file {0}, {1:?}")]
+    ReadFile(String, #[source] std::io::Error),
+    #[error("Can not open from file {0}, {1:?}")]
+    OpenFile(String, #[source] std::io::Error),
+    #[error("Can not parse CPU info, {0:?}")]
+    ParseCpuInfo(#[from] crate::cpu::Error),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+// global config in UT
+#[cfg(test)]
+lazy_static! {
+    static ref SYS_FS_PREFIX: String = format!(
+        "{}/test/texture",
+        std::env::var("CARGO_MANIFEST_DIR").unwrap()
```

Comment:
I think you want `env!` rather than `std::env:var` if you want the compile time value of `CARGO_MANIFEST_DIR`.

Also this would probably be better as a `Path` rather than a `String`.

Reply:
Fixed. But `SYS_FS_PREFIX` can't be converted into PathBuf.

Reply:
fixed